src/app/components/progress-indicator/waterfall-progress-indicator/index.html

```diff
@@ -0,0 +1,325 @@
+<sky-tabbed-demo-page
+  pageTitle="Waterfall Progress Indicator"
+>
+
+  <sky-demo-page-summary>
+    <p>
+      The progress indicator module visually represents progress through sequential steps toward a goal. Waterfall progress indicators walk users through discrete steps on a page.
+    </p>
+  </sky-demo-page-summary>
+
+  <div class="sky-margin-stacked-content-section">
+    <sky-demo-page-example>
+      <sky-progress-indicator-demo>
+      </sky-progress-indicator-demo>
+    </sky-demo-page-example>
+  </div>
+
+  <sky-tabset id="content-tabs">
+    
+    <sky-tab
+      tabHeading="Design">
+    
+      <sky-demo-page-content sectionHeading="Usage">
+        <h3 class="sky-text-success">Use when</h3>
+        <sky-docs-visual-example>
+          <sky-docs-visual-example-text>
+            <p>Use waterfall progress indicators for long or complex setup processes that users typically complete only once. Waterfall indicators are ideal when events outside of the system force users to complete tasks in multiple sittings or when users must enter large amounts of data.</p>
+            <p>Use waterfall progress indicators when processes include at least three steps and the number and order of steps do not change.</p>
+          </sky-docs-visual-example-text>
+          <sky-docs-visual-example-image>
+            <div>
+              <img src="~/assets/img/guidelines/progress-indicator/waterfall-progress-usage-1.png" />
+              <div class="caption caption-success sky-text-success">Do use waterfall progress indicators for complex configuration processes that users may complete in stages.</div>
+            </div>
+          </sky-docs-visual-example-image>
+        </sky-docs-visual-example>
+        
+        <sky-docs-visual-example>
+          <sky-docs-visual-example-text>
+            <p>Use waterfall progress indicators when the complexity of steps varies drastically. For example, one step may require a full-screen modal for complex interactions, while another step requires a small modal for a few simple inputs. Waterfall steps can open <a routerLink="/components/wizard">wizards</a> if necessary.</p>
+          </sky-docs-visual-example-text>
+          <sky-docs-visual-example-image>
+            <div class="sky-margin-stacked-separate">
+              <img src="~/assets/img/guidelines/progress-indicator/waterfall-progress-usage-2.png" />
+              <div class="caption caption-success sky-text-success">Do use modals of the appropriate size for the interactions in each waterfall step.</div>
+            </div>
+            <div>
+              <img src="~/assets/img/guidelines/progress-indicator/waterfall-progress-usage-3.png" />
+              <div class="caption caption-danger sky-text-danger">Don't put inputs inside waterfall progress indicators.</div>
```

Comment:
Should this be a separate Don’t use when item below instead of a secondary image/caption here? I can definitely see where it’s tangential to the Use when info, but it seems to me like it’s distinct enough to warrant it’s own Don’t use when item below. (Either that or a separate Use when item here that says to use waterfall progress indicators when the actions that users need to perform are going to be performed in inputs on modals that are opened from the waterfall indicator but not by inputs directly on the waterfall indicator.)

Reply:
Ported over to #625.